src/webapp/components/dataset-wizard/SummaryDataSet.tsx

```diff
@@ -121,43 +154,91 @@ const useGetMissingSelectedCompanion = (props: { dataSet: DataSet }) => {
         () =>
             _(dataSet.indicators)
                 .filter(indicator => Boolean(indicator.code))
-                .keyBy(indicator => indicator.code),
+                .keyBy(indicator => toLowercaseString(indicator.code)),
         [dataSet]
     );
 
-    const missingSelectedCompanion = React.useMemo(() => {
-        return _(dataSet.indicators)
-            .filter(indicator => indicator.getCompanionCodesFromRules().length > 0)
-            .compactMap(indicator => {
-                const { outcomeRuleIsValid, outputRuleIsValid } =
+    const missingSelectedCompanion: Maybe<MissingCompanionIndicator> = React.useMemo(() => {
+        const allIndicatorValidations = _(dataSet.indicators)
+            .filter(indicator => indicator.getAllCompanionCodesFromRules().length > 0)
+            .map(indicator => {
+                const { outcomeRulesValid, outputRulesValid } =
                     indicator.validateCompanionRules(indicatorByCodes);
 
-                if (outcomeRuleIsValid && outputRuleIsValid) return undefined;
-
-                const { outcomeMessage, outputMessage } = indicator.buildCompanionRuleMessage();
-
-                return {
-                    id: indicator.id,
-                    name: indicator.name,
-                    code: indicator.code,
-                    outcomeMessage: !outcomeRuleIsValid ? outcomeMessage : "",
-                    outputMessage: !outputRuleIsValid ? outputMessage : "",
-                };
+                const { outcomeMessages, outputMessages } = indicator.buildCompanionRuleMessage();
+
+                return [
+                    ...outcomeRulesValid.toPairs().map(
+                        ([scope, isValid]): ValidationItem => ({
+                            code: indicator.code,
+                            scope,
+                            type: "outcomes" as const,
+                            isValid,
+                            message: buildValidationMessages(
+                                isValid,
+                                outcomeMessages.get(scope),
+                                "outcomes"
+                            ),
+                        })
+                    ),
+                    ...outputRulesValid.toPairs().map(
+                        ([scope, isValid]): ValidationItem => ({
+                            code: indicator.code,
+                            scope,
+                            type: "outputs" as const,
+                            isValid,
+                            message: buildValidationMessages(
```

Comment:
Similar logic for creating a `ValidationItem`. Can we extract into a function? buildValidationItemByType(type: "outcome" | "outputs")